File: problemas/iniciante/1038/1038.c
#include <stdio.h>

int main()
{
    double total;
    int codigo, quantidade;

    scanf("%d %d", &codigo, &quantidade);

    switch (codigo)
    {
    case 1:
        total = quantidade * 4.00;
        break;
    case 2:
        total = quantidade * 4.50;
        break;
    case 3:
        total = quantidade * 5.00;
        break;
    case 4:
        total = quantidade * 2.00;
        break;
    case 5:
        total = quantidade * 1.50;
        break;
    }

    printf("Total: R$ %.2lf\n", total);

    return 0;
}

File: problemas/iniciante/1038/1038.hs
import Text.Printf

achaPreco :: [Double] -> Int -> Double
achaPreco [] x = 0
achaPreco (preco:precos) 1 = preco
achaPreco (preco:precos) x = achaPreco precos (x - 1)

main :: IO ()
main = do
    entrada <- getLine
    let precos = [4.00, 4.50, 5.00, 2.00, 1.50]
    let [codigo, quantidade] = map read (words entrada) :: [Int]

    let preco = achaPreco precos codigo
    printf "Total: R$ %.2f\n" (fromIntegral quantidade * preco)